app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -83,6 +83,9 @@ def edit
         if page_is_locked?
           flash[:notice] = Alchemy.t('This page is locked', name: @page.locker_name)
           redirect_to admin_pages_path
+        elsif !@page.editable_by?(current_alchemy_user)
+          flash[:notice] = Alchemy.t('Your user role does not allow you to edit this page.')
+          redirect_to admin_pages_path
```

Comment:
If, because of Alchemy's redirecting behaviour, we redirect to the Dashboard before we get here, I think this branch is never executed, is it? In that case it has to go away just like the spec we deleted.